### ISSUE TITLE
Apiary Recipe Deletes Bees

### ISSUE THREAD

**Issue by ThatGravyboat** [NillerMedDild/Enigmatica6#2303]
**Modpack version**
0.4.10 possibly newer version have it too as the bug is still in the GitHub files.

**Describe the bug**
Resourceful Bee Apiary deletes bees when making it via the recipe while the base mods hive recipes don't this is because the apiary recipes were edited and don't use the Resourceful Bees hive upgrade recipe type. 
As seen here you just do a normal shaped recipe https://github.com/NillerMedDild/Enigmatica6/blob/master/kubejs/server_scripts/enigmatica/kubejs/base/recipes/shaped.js#L886
While Resourceful Bees does https://github.com/Resourceful-Bees/ResourcefulBees/blob/1.16.5/src/main/resources/data/resourcefulbees/recipes/t1_apiary.json which makes it so that it will return a bee box with all the bees in the hives.

**Logs**
N/A

**Screenshots**
N/A

**Comment by TemperedFool** [comment]
@ThatGravyboat thanks for the report... this is my bug for sure. So this is a custom recipe type? I'm planning to move stuff around anyway (pushing the apiary recipes to Normal so I can expert them), so I will play with this later (probably Tuesday given the long weekend).

I'm assuming registering a custom recipe of type "resourcefulbees:hive_upgrade_recipe" will do some magic and make the appropriate crafting table recipes? If so it should be an easy fix. Do you normally do the same thing for wild hives? If so I've overridden those too and will fix them at the same time.

**Comment by ThatGravyboat** [comment]
yeah we check the items if the recipe type is "resourcefulbees:hive_upgrade_recipe" if they have our bees list in them and then add them to a list and return a box with them when the item is crafted so it should work for a nest

**Comment by TemperedFool** [comment]
Sweet, should be a simple fix then. I'll implement it and test on Tuesday.

On the plus side, this small fiasco reminds me that I forgot to override the recipes for the in-world upgrades for Expert...

**Comment by TemperedFool** [comment]
@ThatGravyboat Swapping to a custom "resourcefulbees:hive_upgrade_recipe" recipe did the trick for crafting the T1 apiary from a T4 hive, but I want to confirm behavior with T2-4 apiaries. When using either the default recipes provided by RB, or a custom recipe with the upgrade type, I don't seem to get bees back. Is this expected? 

I wouldn't be surprised given how easy it is to remove bees from an apiary, so if that is expected I should be good. It is a mild issue for the recipes I had set up - they are currently also integrating T4 hives to get the cost in the ballpark I'm looking for. Worst case is the apiaries end up a touch cheaper, which isn't much of an issue imo.

**Comment by TemperedFool** [comment]
Per our discord discussion Gravy, my PR should move everything over to your custom recipe handler. I made the executive decision to leave out the NBT - although I tested and it works - as I'm wary of the long-term impact on RS autocrafting.

Since it sounded like y'all will fix it in the RB end, should be little harm. I also added tool tips and descriptions as warnings to cover our player's bases.
